src/lipservice/src/LEAPR.hpp

```diff
@@ -0,0 +1,224 @@
+  JSON = { 
+    { "delta", card11.delta.value },
+  };
+}
+
+inline void to_json( nlohmann::json& JSON, const LEAPR::Card12& card12 ){
+  JSON = { 
+    { "rho", card12.rho.value }
+  };
+}
+
+inline void to_json( nlohmann::json& JSON, const LEAPR::Card13& card13 ){
+  JSON = { 
+    { "twt",   card13.twt.value },
+    { "c",     card13.c.value },
+    { "tbeta", card13.tbeta.value }
+  };
+}
+
+
+inline void to_json( nlohmann::json& JSON, 
+              const std::optional<LEAPR::OscillatorTuple>& oscillatorTuple ){
+  if (oscillatorTuple){
+    JSON = { 
+      { "energies", std::get<0>(*oscillatorTuple).oe.value },
+      { "weights" , std::get<1>(*oscillatorTuple).ow.value }
+    };
+  }
+  else {
+    JSON =  nlohmann::json:: array();
+  } 
+}
+
+
+
+inline void to_json( nlohmann::json& JSON, 
+          const std::optional<LEAPR::PairCorrelTuple>& pairCorrelationTuple){
+  if (pairCorrelationTuple){
+    JSON = { 
+      { "dka"   , std::get<0>(*pairCorrelationTuple).dka.value },
+      { "skappa", std::get<1>(*pairCorrelationTuple).skappa.value }
+    };
+  }
+  else {
+    JSON = { nlohmann::json::array() };
+  }
+}
+
+
+
+inline nlohmann::json make_json( const LEAPR::TempLoopIter& temp ){
+    nlohmann::json JSON = {
+      { "temperature", std::get<0>(temp).temp.value } // Card 10
+    };
+
+    auto tempTuple = std::get<1>(temp);
+    if (tempTuple){
+      JSON.update(std::get<0>(*tempTuple)); //Card 11
+      JSON.update(std::get<1>(*tempTuple)); //Card 12
+      JSON.update(std::get<2>(*tempTuple)); //Card 13
+      auto oscillatorTuple = std::get<4>(*tempTuple); // Cards 14 & 15
+      JSON["oscillators"] = oscillatorTuple;
+      auto pairCorrelationTuple = std::get<5>( *tempTuple ); // Cards 16 & 17
+      JSON["pairCorrelation"] = pairCorrelationTuple;
+    }
+    return JSON;
+
+}
+
+inline void to_json( nlohmann::json& JSON, const LEAPR::TempLoop& tempLoop){
+  auto temperatures = nlohmann::json::array();
+  for (auto& tempLoop_iter : tempLoop){
+      auto thisTemp = std::get<0>(tempLoop_iter).temp.value;
+      if (thisTemp > 0){
+        temperatures.push_back(make_json(tempLoop_iter));
+      }
+      else {
+        auto lastTemperatureInfo = temperatures[temperatures.size()-1];
+        lastTemperatureInfo["temperature"] = std::abs(thisTemp);
+        temperatures.push_back(lastTemperatureInfo);
+      }
+  }
+  JSON = { { "temperatures", temperatures } };
+}
+
+
+inline void to_json( nlohmann::json& JSON, 
+                     const std::optional<LEAPR::Card19>& card19 ){
+  if ( card19 ){
+    JSON = card19->cfrac.value;
+  }
+  else {
+    JSON = nlohmann::json::array();
+  }
+}
+
+inline void to_json( nlohmann::json& JSON, const std::vector<LEAPR::Card20>& card20List ){
+  if ( card20List.size() > 0 ){
+    std::vector<std::string> card20Values(card20List.size());
+    for (size_t i = 0; i < card20List.size(); ++i){
+      card20Values[i] = *(card20List[i].comment.value);
+    }
```

Comment:
range-based for loops are easier than index-based loops.
```suggestion
    for( const auto& card : card20List ){
      card20Values.push_back( *(card.comment.value) );
    }
```